R/surv-concordance.R

```diff
@@ -0,0 +1,113 @@
+#' @template return
+#' @details
+#'
+#' The concordance index is defined as the proportion of all comparable pairs in
+#' which the predictions and outcomes are concordant.
+#'
+#' Two observations are comparable if:
+#'
+#' 1. both of the observations experienced an event (at different times), or
+#' 2. the observation with the shorter observed survival time experienced an
+#'    event, in which case the event-free subject “outlived” the other.
+#'
+#' A pair is not comparable if they experienced events at the same time.
+#'
+#' Concordance intuitively means that two samples were ordered correctly by the
+#' model. More specifically, two samples are concordant, if the one with a
+#' higher estimated risk score has a shorter actual survival time.
+#'
+#' Larger values of the score are associated with better model performance.
+#'
+#' @inheritParams pr_auc
+#'
+#' @param data A `data.frame` containing the columns specified by `truth` and
+#'   `estimate`.
+#'
+#' @param truth The column identifier for the true class survival result (that
+#'   is created using [survival::Surv()].). This should be an unquoted column
+#'   name although this argument is passed by expression and supports
+#'   [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#'   `_vec()` functions, an [survival::Surv()] object.
+#'
+#' @param estimate The column identifier for the survival probabilities. This
+#'   needs to be a numeric vector.This should be an unquoted column name
+#'   although this argument is passed by expression and supports
+#'   [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#'   `_vec()` functions, a numeric vector.
+#'
+#' @param ... Not currently used.
+#'
+#' @author Emil Hvitfeldt
+#'
+#' @references
+#'   Harrell, F.E., Califf, R.M., Pryor, D.B., Lee, K.L., Rosati, R.A,
+#'   “Multivariable prognostic models: issues in developing models, evaluating
+#'   assumptions and adequacy, and measuring and reducing errors”, Statistics in
+#'   Medicine, 15(4), 361-87, 1996.
+#'
+#' @examples
+#' concordance_survival(data = lung_surv, truth = surv_obj, estimate = age)
+#' @export
+concordance_survival <- function(data, ...) {
+  UseMethod("concordance_survival")
+}
+
+concordance_survival <- new_static_survival_metric(
+  concordance_survival,
+  direction = "maximize"
+)
+
+#' @rdname concordance_survival
+#' @export
+concordance_survival.data.frame <- function(data,
+                                            truth,
+                                            estimate,
+                                            na_rm = TRUE,
+                                            case_weights = NULL,
+                                            ...) {
+  static_survival_metric_summarizer(
+    name = "concordance_survival",
+    fn = concordance_survival_vec,
+    data = data,
+    truth = !!enquo(truth),
+    estimate = !!enquo(estimate),
+    na_rm = na_rm,
+    case_weights = !!enquo(case_weights)
+  )
```

Comment:
Is the idea again to rely on `static_survival_metric_summarizer()` to check the dots for us? If so, then we should pass those on here.

Reply:
Good spotted!

Reply:
Turns out this was correct all along. It is an old design decision we are stuck in, but we need to not pass in `...` to make metric_set()` work